### ISSUE TITLE
Improve support for wildcard routes

### ISSUE THREAD

**Issue by ddeaguiar** [pedestal/pedestal#711]
# Description
While the linear search router provides service developers with more controls over routing, The amount of control depends on the route syntax used. When using wildcard routes in conjunction with the set-based syntax for table routes, control over route order is lost. The results can be surprising to service developers. While this can be mitigated by explicitly performing route expansion using the `table/table-routes` fn, having to do so is unclear. This ticket explores preferring static routes over wildcard routes, so that explicit route expansion is no longer necessary.

See [this](https://github.com/pedestal/pedestal/issues/709#issuecomment-1036354811) comment in #709

**Comment by ddeaguiar** [comment]
We could document the idiosyncrasies described in the description but preferring static routes over wildcard routes seems like a sensible approach.

**Comment by hlship** [comment]
Certainly, we are getting away from explicit expansion of routes, as that interferes with a REPL-reloaded approach.

I believe that the new Sawtooth router will have fewer conflicts, as it does a better job of comparing the number of elements in a route URL than prefix-tree, AFAIK.

But I also wonder if there might be a solution that ranks possibly conflicting routes, so that you could match "/*path" for resources, but prefer to match "/api/foo/:bar" even though those are technically in conflict.

Possibly any path-matching wildcard should simply be lower-priority than any standard (non-path-wildcard) route.  When Sawtooth is dispatching, it could consider all standard routes that may match before any wildcard routes.
